### PR TITLE
[Doc] - Meshery installation documentation update

### LINKED ISSUE #6629: [Docs] Improve Setup Installation
#### Current State

Currently, meshery docs don't tackle all the different use cases that a user might get into while starting Meshery for the first time. Some challenges include network challenges with the K8s cluster of `minikube` and `kind` which require loadbalancer setup for the Meshery Services to be exposed successfully.

#### Desired State

Improve Meshery Docs for better setup instructions. 
Eg: Improve Meshery installation with K8s clusters of type [Minikube](https://docs.meshery.io/installation/platforms/minikube) and [KinD](https://docs.meshery.io/installation/platforms/kind)

Related to: https://github.com/meshery/meshery/issues/6621

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by gurpreet-legend**
Signed-off-by: Gurpreet Singh <lostlegend2701@gmail.com>

Updated the installation troubleshooting documentation as per the discussion in #6621 

This PR fixes #6629 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by gurpreet-legend** [comment]
@meshery/meshery-docs-contributors, can someone review this?

**Comment by l5io** [comment]
🚀 Preview for commit 0b659c10da3756e5915596bb10d0669e9f246c13 at: https://63bc97aa5546a2054e823e74--meshery-docs-preview.netlify.app

File: docs/pages/reference/troubleshooting/installation.md [hunk 1]
```diff
@@ -20,6 +20,25 @@ Anytime a `mesheryctl system` command is executed, a series of preflight checks
...
+
+2. Use minikube tunnel to expose services. `minikube tunnel`. 
+
+    Docs: [link](https://minikube.sigs.k8s.io/docs/handbook/accessing/#using-minikube-tunnel)
+
+    A more simple way to resolve this issue can be `port-forwarding`. Run the following command in terminal:
+
+    `kubectl port-forward service/meshery 9081:9081 -n meshery`
+
+3. For `kind`, I prefer installing MetalLB with a custom configmap.
```
Comment (leecalcote) [inline docs/pages/reference/troubleshooting/installation.md:38]: Let's avoid first-person statements in Meshery Docs, using active voice instead; e.g. "You can install MetalLB with a custom ConfigMap".

**Review by leecalcote — CHANGES_REQUESTED** [review]
Thanks, @gurpreet-legend. Will you undo changes to the older version(s) of the docs, leaving changes only to the current / latest version?

File: docs/pages/reference/troubleshooting/installation.md [hunk 2]
```diff
@@ -20,6 +20,25 @@ Anytime a `mesheryctl system` command is executed, a series of preflight checks
...
+The difficulty with Minikube and Kind clusters is that they typically don't have a loadbalancer, whereas Meshery gives its users complete control over how networking is configured. As a result, networking for these deployments is more complex. Few of the alternatives are:
+1. Use the MetalLB minikube add-on that provides load balancing. `minikube addons enable metallb`
+
+    MetalLB setup: [link](https://kubebyexample.com/learning-paths/metallb/install)
+
+2. Use minikube tunnel to expose services. `minikube tunnel`. 
+
+    Docs: [link](https://minikube.sigs.k8s.io/docs/handbook/accessing/#using-minikube-tunnel)
+
+    A more simple way to resolve this issue can be `port-forwarding`. Run the following command in terminal:
```
Comment (leecalcote) [inline docs/pages/reference/troubleshooting/installation.md:34]: "more simple" change to "simpler"

File: docs/pages/reference/troubleshooting/installation.md [hunk 3]
```diff
@@ -20,6 +20,25 @@ Anytime a `mesheryctl system` command is executed, a series of preflight checks
 
     Situation: Unable to start Meshery Server with `make run-local` due to error of `key/value size is invalid`
 
+## Setting up Meshery using Kind or Minikube
+
+The difficulty with Minikube and Kind clusters is that they typically don't have a loadbalancer, whereas Meshery gives its users complete control over how networking is configured. As a result, networking for these deployments is more complex. Few of the alternatives are:
```
Comment (leecalcote) [inline docs/pages/reference/troubleshooting/installation.md:25]: If "loadbalancer" in this sense refers to the Kubernetes construct, please capitalize the "L" and "B".

File: docs/v0.5/guides/troubleshooting/installation.html [hunk 4]
```diff
@@ -599,6 +599,27 @@ <h2 id="mesherys-preflight-checks">Meshery’s Preflight Checks</h2>
   </li>
 </ol>
 
+<h2 id="setting-up-meshery-using-kind-or-minikube">Setting up Meshery using Kind or Minikube</h2>
+
+<p>The difficulty with Minikube and Kind clusters is that they typically don't have a loadbalancer, whereas Meshery gives its users complete control over how networking is configured. As a result, networking for these deployments is more complex. Few of the alternatives are:</p>
+
+<ol>
+  <li>
+    <p>Use the MetalLB minikube add-on that provides load balancing.<code class="language-plaintext highlighter-rouge">minikube addons enable metallb</code></p>
```
Comment (leecalcote) [inline docs/v0.5/guides/troubleshooting/installation.html:608]: Please capitalize the "M" in Minikube

Comment (leecalcote) [reply docs/v0.5/guides/troubleshooting/installation.html:608]: ...for all instances of Minikube that are not CLI references

**Comment by gurpreet-legend** [comment]
@leecalcote , all the required changes are done.

**Comment by l5io** [comment]
🚀 Preview for commit e30590abe334342b16e3365814b38f10a103abbf at: https://63bd049fcf0769583d01886b--meshery-docs-preview.netlify.app

**Comment by gurpreet-legend** [comment]
> 

@Chadha93, sure will discuss this at the meet then :)

File: docs/pages/reference/troubleshooting/installation.md [hunk 5]
```diff
@@ -20,6 +20,25 @@ Anytime a `mesheryctl system` command is executed, a series of preflight checks
 
     Situation: Unable to start Meshery Server with `make run-local` due to error of `key/value size is invalid`
 
+## Setting up Meshery using Kind or Minikube
+
+The difficulty with Minikube and Kind clusters is that they typically don't have a LoadBalancer, whereas Meshery gives its users complete control over how networking is configured. As a result, networking for these deployments is more complex. Few of the alternatives are:
```
Comment (leecalcote) [inline docs/pages/reference/troubleshooting/installation.md:25]:
```suggestion
The difficulty with Minikube and Kind clusters is that they typically don't support LoadBalancer service networking by default. Meshery UI and Meshery Broker are configured for LoadBalancer service networking by default. There are a number of solutions this overcoming this challenge. Here are a few methods:
```

**Review by leecalcote — COMMENTED** [review]
Excellent.

**Review by leecalcote — APPROVED** [review]
Tweaked the language a little.

LGTM 👍

**Pull request merged**